irohad/model/sha3_hash.cpp

```diff
@@ -33,7 +33,7 @@ namespace iroha {
 
   hash256_t hash(const model::Block &block) {
     auto &&pb_dat = block_factory.serialize(block);
-    return hash(pb_dat);
+    return hash(pb_dat.block_v1());
```

Comment:
Probably we have to expect here appearing of some other block versions rather than v1.
Should we print something if some other version of a block was passed here?

Reply:
If some fix is possible and needed here, please do the same fix for all other applicable places.